presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -439,6 +527,16 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
         return builder.build();
     }
 
+    /**
+     * Sets a default for serializedData and transforms ir to json.
```

Comment:
- typo;
- this method extracts the PRESTO_QUERY_REQUEST_METADATA header, parses escaped slashes and double quotes, than deserialize to QueryRequestMetadata with a json codec.